File: app/src/main/java/com/michelbarbosa/hsdm_hearthstonedustmanager/ui/activity/YourDecksActivity.java
package com.michelbarbosa.hsdm_hearthstonedustmanager.ui.activity;

import android.app.AlertDialog;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.michelbarbosa.hsdm_hearthstonedustmanager.R;
import com.michelbarbosa.hsdm_hearthstonedustmanager.enums.DialogType;
import com.michelbarbosa.hsdm_hearthstonedustmanager.ui.components.CustomDialog;
import com.michelbarbosa.hsdm_hearthstonedustmanager.ui.interfaces.RemoteConfigCompleteListener;
import com.michelbarbosa.hsdm_hearthstonedustmanager.utils.RemoteConfigUtils;
import com.michelbarbosa.hsdm_hearthstonedustmanager.utils.UIUtil;

public class YourDecksActivity extends MainActivity implements RemoteConfigCompleteListener {

    private double currentVersion;
    private String updateUrl;
    private boolean updateRequired, dataReloadRequired;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setLayoutContent(R.layout.activity_yourdecks);
        setToolbarTitle(R.string.title_yourDecks);

        RemoteConfigUtils.INSTANCE.startCallback(this);

        Button btCreateDeck = findViewById(R.id.bt_yourdecks_createDeck);
        btCreateDeck.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                gotoCreateDeckActivity(YourDecksActivity.this);
            }
        });
    }

    public void showDialog() {
        final AlertDialog progressDialog = UIUtil.progressDialog(this, "Verificando atualizações...");
        progressDialog.show();

        final CustomDialog testeDialog = new CustomDialog(this,
                "currentVersion -" + currentVersion + "\n"
                        + "updateURL - " + updateUrl + "\n"
                        + "updateRequired? - " + updateRequired + "\n"
                        + "dataReloadRequired? - " + dataReloadRequired
                , DialogType.INFO
                , false
        );
        testeDialog.show();
        testeDialog.onOptionConfirmClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                testeDialog.dismiss();
            }
        }, "OK");

        progressDialog.dismiss();
    }

    @Override
    public void onComplete() {
        currentVersion = RemoteConfigUtils.INSTANCE.getCurrentVersion();
        updateUrl = RemoteConfigUtils.INSTANCE.getUpdateUrl();
        updateRequired = RemoteConfigUtils.INSTANCE.getUpdateRequired();
        dataReloadRequired = RemoteConfigUtils.INSTANCE.getDataReloadRequired();

        showDialog();
    }

}

File: KOTLIN_MIGRATION/HDM-Hearthstone_Dust_Manager/app/src/main/java/com/michelbarbosa/hdm_hearthstone_dust_manager/ui/activities/YourDecksActivity.kt
package com.michelbarbosa.hdm_hearthstone_dust_manager.ui.activities

import android.os.Bundle
import com.michelbarbosa.hdm_hearthstone_dust_manager.R
import com.michelbarbosa.hdm_hearthstone_dust_manager.utils.remoteconfig.RemoteConfigCompleteListener
import com.michelbarbosa.hdm_hearthstone_dust_manager.utils.remoteconfig.RemoteConfigUtils
import kotlinx.android.synthetic.main.activity_yourdecks.*
import michel566.androidmodules.lightdialog.DialogType
import michel566.androidmodules.lightdialog.LightDialog

/**
 * Created by Michel Barbosa on 29/07/2021.
 */
class YourDecksActivity : MainActivity(), RemoteConfigCompleteListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setLayoutContent(R.layout.activity_yourdecks)
        setToolbarTitle(R.string.title_yourDecks)

        initialize()
        setupWidgets()
    }

    private fun initialize() {
        RemoteConfigUtils.startCallback(this)
    }

    private fun setupWidgets() {
        bt_yourdecks_createDeck.setOnClickListener {
            gotoCreateDeckActivity()
        }
    }

    override fun onComplete() {
        val checkDialogUpdate = LightDialog(
            this,
            "current version - " + RemoteConfigUtils.getCurrentVersion() + "\n"
                    + "updateURL - " + RemoteConfigUtils.getUpdateUrl() + "\n"
                    + "updateRequired - " + RemoteConfigUtils.getUpdateRequired() + "\n"
                    + "dataReloadRequired - " + RemoteConfigUtils.getDataReloadRequired(),
            DialogType.INFO, false
        )
        checkDialogUpdate.show()
        checkDialogUpdate.onOptionConfirmClickListener({
            checkDialogUpdate.dismiss()
        }, "OK")
    }
}